### ISSUE TITLE
Node20 Externals Version needs upgrade [CVE-2023-32002]

### ISSUE THREAD

**Issue by mondo192** [actions/runner#2902]
**Describe the bug**
Security scanning of the default installation method results in [CVE-2023-32002](https://github.com/advisories/GHSA-9m48-r3w4-x35v) being tripped based on the current version of Node JS 20 set in externals.

**To Reproduce**
1. Take latest installation from releases including runtimes and externals. Example: actions-runner-linux-x64-2.309.0.tar.gz
2. Uncompress
3. Run security scan (e.g. Wiz)

**Expected behavior**
Clean security report

## Runner Version and Platform
v2.309.0/v2.310.2

OS of the machine running the runner? Linux

## What's not working?
```
File /home/runner/runner/agent/externals/node20/bin/node version 20.5.0 is vulnerable to CVE-2023-32002, which exists in versions >= 20.0.0, <= 20.5.0.

The vulnerability was found in the [National Vulnerability Database (NVD)](https://nvd.nist.gov/vuln/detail/CVE-2023-32002) based on the CPE cpe:2.3:a:nodejs:node.js with NVD severity: Critical.

The file is associated with the technology Node.js.

The vulnerability can be remediated by updating Node.js to 20.5.1 or higher.
```

**Comment by wchesley** [comment]
I can confirm this still exists in v2.310.2, below is an excerpt from a Nessus scan ran this morning. 

```  
Path: ~/actions-runner/externals.2.310.2/node20/bin/node
  Installed version : 20.5.0
  Fixed version     : 20.5.1

  Path: ~/actions-runner/externals.2.309.0/node20/bin/node
  Installed version : 20.5.0
  Fixed version     : 20.5.1
```

**Comment by MikeMcC399** [comment]
should be fixed now

https://github.com/actions/runner/blob/121f080023450ab0ee5fc2d7961d5b630434d41f/src/Misc/externals.sh#L9
